src/interfaces/background_field_interface.jl

```diff
@@ -0,0 +1,198 @@
+####################
+# The abstract background field interface
+#
+# In this file, the abstract interfaces for different types of background fields
+# is defined. 
+####################
+"""
+Abstract base type for describing classical background fields.
+
+The only supported class of background fields yet, are the pulsed plane-wave field (see [`AbstractPulsedPlaneWaveField`](@ref) for details).
+"""
+abstract type AbstractBackgroundField end
+
+#####
+# pulsed plane-wave fields
+#####
+
+"""
+Abstract base type for pulsed plane-wave fields.
+
+!!! note "Pulsed field interface"
+
+    For every subtype, the following functions need to be implemented.
+
+    ```Julia
+
+    reference_momentum(::AbstractPulsedPlaneWaveField)
+    domain(::AbstractPulsedPlaneWaveField)
+    phase_duration(::AbstractPulsedPlaneWaveField)
+    envelope(::AbstractPulsedPlaneWaveField,x::Real)
+    ```
+
+"""
+abstract type AbstractPulsedPlaneWaveField <: AbstractBackgroundField end
+
+"""
+
+    reference_momentum(::AbstractPulsedPlaneWaveField)
+
+Interface function for [`AbstractPulsedPlaneWaveField`](@ref), which returns the reference momentum (as a subtype of `QEDbase.AbstractFourMomentum`) associated with the passed background field.
+"""
+function reference_momentum end
+
+"""
+    domain(::AbstractPulsedPlaneWaveField)
+
+Interface function for [`AbstractPulsedPlaneWaveField`](@ref), which returns interval (as a `IntervalSets.Interval`) for the given background field. 
+
+"""
+function domain end
+
+"""
+    
+    phase_duration(::AbstractPulsedPlaneWaveField)
+
+Interface function for [`AbstractPulsedPlaneWaveField`](@ref), which returns the phase extent of the background field.
+"""
+function phase_duration end
+
+"""
+
+    _envelope(::AbstractPulsedPlaneWaveField, phi::Real)
+
+Interface function for [`AbstractPulsedPlaneWaveField`](@ref), which returns the value of the phase envelope function (also referred to as pulse envelope) for a given phase variable.
+
+!!! note "Single point implementation"
```

Comment:
Is this note still required with the existing implementation in line 85?

Reply:
IMHO, this is required because the user will not read the code of line 85 if he or she implements the interface. Therefore, this is a hint, that implementing `_envelope(field,phi::Real)` is enough to get the naive implementation of `_envelope(field,phi::AbstactVector{Real})` just for free.